### PR TITLE
[FLINK-31046][FLINK-31047][Kubernetes-Operator] - Improvement/enhance Flink operator template YAML

### PR DESCRIPTION
**Pull request by tamirsagi**
## What is the purpose of the change

This PR adds support for
- Extra environment variables which are available upon deployment (hostIP, podIP, podName)
- Support pod topology constrains in case number of Replicas > 1

## Brief change log

  - Change flink-operator template yaml to provide such environments variables and topology.
 - Move serviceAccount field to valid location in pod-template example

## Documentation

- Update content/docs/operations/configuration.md

### CODE REVIEW

**Comment by tamirsagi** [comment]
@gyfora  Would you please review?

File: docs/content/docs/operations/configuration.md [hunk 1]
```diff
@@ -81,6 +81,19 @@ Lease name must be unique in the current lease namespace. For other more advance
...
+
+
+## Environment variables
+The operator exposes several environment variables which can be used for custom plugins.
+
+| Name     | Description                           | FieldRef      |
+|----------|---------------------------------------|---------------|
+| HOST_IP  | The host which the pod is deployed on | status.hostIP |
+| POD_IP   | Pod IP                                | status.podIP  |
+| POD_NAME | Pod Name                              | matadata.name |
```
Comment (john-coleman) [inline docs/content/docs/operations/configuration.md:95]:
```suggestion
| POD_NAME | Pod Name                              | metadata.name |
```

File: docs/content/docs/custom-resource/pod-template.md [hunk 2]
```diff
@@ -53,13 +53,13 @@ spec:
...
   flinkConfiguration:
     taskmanager.numberOfTaskSlots: "2"
+  serviceAccount: flink
   podTemplate:
     apiVersion: v1
     kind: Pod
     metadata:
       name: pod-template
     spec:
-      serviceAccount: flink
```
Comment (gyfora) [inline docs/content/docs/custom-resource/pod-template.md:62]: serviceAccount is part of the spec, please dont change this

Comment (tamirsagi) [reply docs/content/docs/custom-resource/pod-template.md:62]: Once its there it fails upon validation.

https://github.com/apache/flink-kubernetes-operator/blob/main/flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java#L523

I had to move it to get the deployment to work.

Comment (gyfora) [reply docs/content/docs/custom-resource/pod-template.md:62]: ah sorry, I looked at your change backwards, this is correct :)

**Review by gyfora — APPROVED** [review]

**Pull request merged**